app/src/components/fields/MultiAutocompleteFieldVariableSize.tsx

```diff
@@ -3,12 +3,14 @@ import ListSubheader from '@material-ui/core/ListSubheader';
 import TextField from '@material-ui/core/TextField';
 import CheckBox from '@material-ui/icons/CheckBox';
 import CheckBoxOutlineBlank from '@material-ui/icons/CheckBoxOutlineBlank';
-import Autocomplete from '@material-ui/lab/Autocomplete';
+import Autocomplete, { AutocompleteInputChangeReason, createFilterOptions } from '@material-ui/lab/Autocomplete';
```

Comment:
(Not specific to this line)

Because this file is fairly complex, compared to most components, it would be worth adding some extra comments wherever you think some extra explanation is needed. Same goes for any functions or variable names (are there any names that can be made more descriptive, etc). Also, any functions/variables that are only used when you are doing an `api-search`, if you think there is any meaningful naming or grouping or comments we can add to them, so that its clear they are only used when in that mode, and not the other.

Reply:
There are probably tons of ways to do this, but the main goal should be clarity for future devs who might have to make a change.  Obv some amount of complexity is unavoidable, but anywhere we can help reduce complexity is worth it.